### DISCUSSION TITLE
How to determine which version of a formula is linked?

### DISCUSSION THREAD

**Discussion by Eydamos** [Homebrew/discussions#1054] [Everyday usage] [answered]
In my case I have multiple php versions installed: php, php@7.4, php@7.2
Is there a way to know which version is currently linked? I need this for an automated script which should iterate over the versions unlink the current version, link the next one and run some scripts with that version.
I could of course just iterate over all versions and do a `brew unlink` before linking the next one but I don't really like this approach as it consumes unnecessary time.

**Comment by carlocab** [comment, marked as answer]
Not sure if this is the best way, but you can try looking in `${HOMEBREW_PREFIX}/var/homebrew/linked`.

**Reply by Eydamos** [reply]
Thank you this already helps. Would be nice to have a command or a command param for that though.

**Comment by gromgit** [comment]
```bash
# linked_formula <symlink>
linked_formula() {
  if [[ $(readlink "$1") =~ .*/Cellar/([^/]*)/.* ]]; then
    echo "${BASH_REMATCH[1]}"
  else
    echo "ERROR: '$1' is not a Cellar symlink." >&2
    return 1
  fi 
}

# Usage...
$ linked_formula /usr/local/bin/php
```
